src/app-logic/browser-connection.js

```diff
@@ -65,40 +59,41 @@ export interface BrowserConnection {
  */
 class BrowserConnectionImpl implements BrowserConnection {
   _webChannelSupportsGetProfileAndSymbolication: boolean;
-  _geckoProfiler: $GeckoProfiler | void;
```

Comment:
I removed the caching of the geckoProfiler object in favor of awaiting the promise always. When the promise is resolved this should be as fast, but asynchronous, which isn't a problem for us because all APIs are asynchronous already.